x-pack/plugins/observability/public/components/shared/exploratory_view/header/embed_action.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButtonEmpty, EuiPopover, EuiCodeBlock, EuiPopoverTitle } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useState } from 'react';
+import { TypedLensByValueInput } from '@kbn/lens-plugin/public';
+import { useSeriesStorage } from '../hooks/use_series_storage';
+
+export function EmbedAction({
+  lensAttributes,
+}: {
+  lensAttributes: TypedLensByValueInput['attributes'] | null;
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const { reportType, allSeries } = useSeriesStorage();
+
+  const button = (
+    <EuiButtonEmpty
+      size="s"
+      isDisabled={lensAttributes === null}
+      onClick={() => {
+        setIsOpen(!isOpen);
+      }}
+    >
+      {EMBED_LABEL}
+    </EuiButtonEmpty>
+  );
+
+  return (
+    <EuiPopover button={button} isOpen={isOpen} closePopover={() => setIsOpen(false)}>
+      <EuiPopoverTitle>{EMBED_TITLE_LABEL}</EuiPopoverTitle>
+      <EuiCodeBlock
+        language="jsx"
+        fontSize="m"
+        paddingSize="m"
+        isCopyable={true}
+        style={{ width: 500 }}
+      >
+        {`const { observability } = useKibana<>().services;
```

Comment:
Not valid TS.

```suggestion
        {`const { observability } = useKibana().services;
```

Reply:
Aha, ok. Perhaps we could include some text in there like `{PLUGIN_START_TYPE}` to highlight what's expected. That might be nicer for the user than having it empty, where they might delete it without realizing the hook requires a generic type arg.